File: snapchat/profile/oc/CodiaDemoOC/ViewController.m
#import "ViewController.h"
#import "ProfileContainer.h"

@interface ViewController ()

@property (nonatomic, strong) ProfileContainer *mainView;
@property (nonatomic, strong) UIScrollView *scrollView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.scrollView = [[UIScrollView alloc] initWithFrame:CGRectMake(0, [self statusBarHeight], self.view.bounds.size.width, self.view.bounds.size.height - [self statusBarHeight])];
    [self.view addSubview:self.scrollView];
    self.mainView = [[ProfileContainer alloc]initWithFrame:CGRectMake(0, 0, 401, 858)];
    self.scrollView.contentSize = [self fitScreen:self.mainView];
    [self.scrollView addSubview:self.mainView];
}


- (CGFloat)statusBarHeight {
    UIWindowScene *windowScene = (UIWindowScene*)[[UIApplication sharedApplication].connectedScenes anyObject];
    return windowScene.statusBarManager.statusBarFrame.size.height;
}

- (CGSize)fitScreen:(UIView *)view {
    CGFloat width = view.frame.size.width;
    CGFloat screenWidth = [UIScreen mainScreen].bounds.size.width;
    if (@available(iOS 16.0, *)) {
        view.anchorPoint = CGPointMake((width / screenWidth) / 2 , (width / screenWidth) / 2);
    } else {
        view.layer.anchorPoint = CGPointMake((width / screenWidth) / 2 , (width / screenWidth) / 2);
    }
    view.transform = CGAffineTransformMakeScale(screenWidth / width, screenWidth / width);
    return CGSizeMake(screenWidth, view.frame.size.height);
}

@end 

File: snapchat/profile/swift/CodiaDemoSwift/ViewController.swift
import UIKit

class ViewController: UIViewController {
    var scrollView: UIScrollView!
    var mainView: UIView!

    override func viewDidLoad() {
        super.viewDidLoad()
        scrollView = UIScrollView(frame: CGRect(x: 0, y: statusBarHeight(), width: view.bounds.width, height: view.bounds.height - statusBarHeight()))
        view.addSubview(scrollView)
        view.backgroundColor = .white
        mainView = ProfileContainer(frame: CGRect(x: 0, y: 0, width: 401, height: 858))
        scrollView.contentSize = fitScreen(mainView)
        scrollView.addSubview(mainView)
    }

    func statusBarHeight() -> CGFloat {
        guard let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene else {
            return 0
        }
        return windowScene.statusBarManager?.statusBarFrame.size.height ?? 0
    }

    func fitScreen(_ view: UIView) -> CGSize {
        let width = view.frame.size.width
        let screenWidth = UIScreen.main.bounds.size.width
        if #available(iOS 16.0, *) {
            view.anchorPoint = CGPoint(x: (width / screenWidth) / 2, y: (width / screenWidth) / 2)
        } else {
            view.layer.anchorPoint = CGPoint(x: (width / screenWidth) / 2, y: (width / screenWidth) / 2)
        }
        view.transform = CGAffineTransform(scaleX: screenWidth / width, y: screenWidth / width)
        return CGSize(width: screenWidth, height: view.frame.size.height)
    }
}
